validator/sawtooth_validator/execution/context_manager.py

```diff
@@ -61,13 +62,18 @@ def __init__(self, state_hash, read_list, write_list, base_context_ids):
             base_context_ids (list of str): Context ids of contexts that this
                 context is based off of.
         """
+
         self._state_hash = state_hash
 
-        self._read_list = set(read_list)
-        self._write_list = set(write_list)
+        self._read_list = list(read_list)
+        self._write_list = list(write_list)
```

Comment:
If those items are passed in via a generator or a list comprehension, this coerces them to be an explicit list.  If it's already a list, its effectively a no-op.